fix(csi/node): patch ana label with merge

tiagolobocastro opened the pull request:
ci(gha): upload test report
    
    Add test script which generates a report tar with useful
    information such as journalctl logs and nvme devices
    Upload this tar on github actions
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    test(dmsetup): ensure lvm device suspended
    
    The slow_pool tests fails on CI because of 2 separate reasons:
    1. we need to wait until device is susppended
    2. we need to disable auto activation of the VG which actives the LVs
    
    Adds a busy-wait until the device is suspended.
    Also setup losetup with direct io on always.
    Adds a bit more logging in case things go awry.
    Disables VG auto activation.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    test: cleanup loop devices properly
    
    The = sign is not part of the jq arg, so remove it.
    Also simplify the whole thing in a single jq command.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(csi/node): patch ana label with merge
    
    Previously a server-side apply was being used which seems to have issues on some k8s
    clusters with dual ip stack.
    Here we're patching a label so a merge patch should suffice.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
fix(csi/node): patch ana label with merge

Previously a server-side apply was being used which seems to have issues on some k8s
clusters with dual ip stack.
Here we're patching a label so a merge patch should suffice.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
Resolves https://github.com/openebs/openebs/issues/3727

tiagolobocastro committed:
test: cleanup loop devices properly

The = sign is not part of the jq arg, so remove it.
Also simplify the whole thing in a single jq command.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

sinhaashish commented on control-plane/csi-driver/src/bin/node/k8s.rs:
```diff
@@ -16,8 +16,8 @@ pub(crate) async fn patch_k8s_node(
     match nodes
         .patch(
             node_name,
-            &PatchParams::apply("node_label_patch").force(),
```
apply was with `force()` . Dont we need that in `Merge` ?

tiagolobocastro replied:
Force is only for apply, see docs for force function:
```rust
/// Force the result through on conflicts
///
/// NB: Force is a concept restricted to the server-side [`Patch::Apply`].
```

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
> Build failed:
> 
>     * [bors-ci](https://github.com/openebs/mayastor-control-plane/actions/runs/12958792707/job/36150493943)

How strange, test failed because the pool creation to a suspended device succeeded!?
Perhaps suspend command is not sync and we have to wait until it's really suspended?
Anyways, not related to this change so let's try again:
bors merge

tiagolobocastro committed:
ci(gha): upload test report

Add test script which generates a report tar with useful
information such as journalctl logs and nvme devices
Upload this tar on github actions

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
Alright, now we wait for device suspended, should do the trick?
bors merge

tiagolobocastro commented:
bors try

tiagolobocastro committed:
test(dmsetup): ensure lvm device suspended

The slow_pool tests fails on CI because of 2 separate reasons:
1. we need to wait until device is susppended
2. we need to disable auto activation of the VG which actives the LVs

Adds a busy-wait until the device is suspended.
Also setup losetup with direct io on always.
Adds a bit more logging in case things go awry.
Disables VG auto activation.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
Turns out the issue was related to VG auto activation, which on CI seems to suffer from delay and run after the lvol is created. The auto activation was activating the suspended lvol :)
Disabling auto-activation on the test fixes this.
I couldn't have figure this out without journalctl inspection, so I'm not generating and uploading a report on failure.
bors merge
